src/plugins/ini/ini.c

```diff
@@ -237,9 +294,81 @@ static void iniBomHandler(void *vhandle, short BOM)
 	}
 }
 
+static int iniKeyToElektraArray(CallbackHandle *handle, Key *existingKey, Key *appendKey, const char *value)
+{
+
+	keySetMeta(appendKey, "ini/section", 0);
+	if (keyGetMeta(existingKey, "ini/array"))
+	{
+		//array already exists, appending new key
+		const char *lastIndex = keyString(keyGetMeta(existingKey, "ini/array"));
+		keyAddBaseName(appendKey, lastIndex);
+		keySetMeta(appendKey, "ini/array", 0);
+		keySetMeta(appendKey, "order", 0);
+		keySetMeta(appendKey, "parent", 0);
+		if (elektraArrayIncName(appendKey) == 1)
+		{
+			return -1;
+		}
+		keySetString(appendKey, value);
+		keySetMeta(appendKey, "ini/key", 0);
+		keySetMeta(appendKey, "ini/arrayMember", "");
+		ksAppendKey(handle->result, appendKey);
+		keySetMeta(existingKey, "ini/array", keyBaseName(appendKey));
+		ksAppendKey(handle->result, existingKey);
+	}
+	else
+	{
+		//creating a new array
+		Key *sectionKey = keyDup(appendKey);
+		keyAddName(sectionKey, "..");
+		char *origVal = strdup(keyString(existingKey));
+		keySetString(appendKey, "");
+		keySetMeta(appendKey, "ini/array", "#1");
+		setSectionNumber(handle->parentKey, appendKey, handle->result);
+		setOrderNumber(handle->parentKey, appendKey);
+		keySetMeta(appendKey, "ini/key", "");
+		keySetMeta(appendKey, "parent", 0);
+		ksAppendKey(handle->result, keyDup(appendKey));
+		keySetMeta(appendKey, "ini/arrayMember", "");
+		keySetMeta(appendKey, "ini/key", 0);
+		keySetMeta(appendKey, "ini/array", 0);
+		keySetMeta(appendKey, "parent", 0);
+		keySetMeta(appendKey, "ini/section", 0);
+		keyAddName(appendKey, "#");
+		keySetMeta(appendKey, "order", 0);
+		if (elektraArrayIncName(appendKey) == -1)
+		{
+			free(origVal);
+			return -1;
+		}
+		keySetString(appendKey, origVal);
+		ksAppendKey(handle->result, keyDup(appendKey));
+		free(origVal);
+		if (elektraArrayIncName(appendKey) == -1)
+		{
+			return -1;
+		}
+		keySetMeta(appendKey, "parent", 0);
+		keySetString(appendKey, value);
+		ksAppendKey(handle->result, keyDup(appendKey));
+		keyDel(appendKey);
+		keyDel(sectionKey);
+	}
+	return 1;
+}
+
 static int iniKeyToElektraKey (void *vhandle, const char *section, const char *name, const char *value, unsigned short lineContinuation)
 {
 	CallbackHandle *handle = (CallbackHandle *)vhandle;
+	if ((!section || *section == '\0') && (!name || *name == '\0'))
+	{
+		keySetString(handle->parentKey, value);
```

Comment:
That is not a good idea, the parentKeys string needs to be untouched, it contains the file to parse.